File: tests/selenium/catroid/LicenseTests.php
<?php
/*    Catroid: An on-device graphical programming language for Android devices
 *    Copyright (C) 2010-2011 The Catroid Team
 *    (<http://code.google.com/p/catroid/wiki/Credits>)
 *
 *    This program is free software: you can redistribute it and/or modify
 *    it under the terms of the GNU Affero General Public License as
 *    published by the Free Software Foundation, either version 3 of the
 *    License, or (at your option) any later version.
 *
 *    This program is distributed in the hope that it will be useful,
 *    but WITHOUT ANY WARRANTY; without even the implied warranty of
 *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *    GNU Affero General Public License for more details.
 *
 *    You should have received a copy of the GNU Affero General Public License
 *    along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */

/* Set TESTS_BASE_PATH in testsBootstrap.php to your catroid www-root */
require_once 'testsBootstrap.php';

class LicenseTests extends PHPUnit_Framework_TestCase
{
  private $selenium;

  public function setUp()
  {
    $path= 'http://'.str_replace('http://', '', TESTS_BASE_PATH).'catroid/';
    $this->selenium = new Testing_Selenium("*firefox", $path);
    if (TESTS_SLOW_MODE==TRUE) {
      $this->selenium->setSpeed(TESTS_SLOW_MODE_SPEED);
    } else {
      $this->selenium->setSpeed(1);
    }
    $this->selenium->start();
  }

  public function tearDown()
  {
    $this->selenium->stop();
  }
  
  public function testImprint() {
    $this->selenium->open(TESTS_BASE_PATH);
    $this->selenium->waitForPageToLoad(10000);
    // Privacy Policy | Terms of Use | Copyright Policy | Imprint | Contact 
 
    // contact us
    $this->selenium->click("xpath=//a[@class='license'][5]");    
    $this->selenium->waitForPageToLoad(10000);
    $this->assertTrue($this->selenium->isTextPresent(("Contact us")));        
    $this->selenium->isElementPresent("xpath=//p[@class='licenseText']/a");
    $this->selenium->goBack();     
    $this->selenium->waitForPageToLoad(10000);
    // copyright policy
    $this->selenium->click("xpath=//a[@class='license'][3]");    
    $this->selenium->waitForPageToLoad(10000);
    $this->assertTrue($this->selenium->isTextPresent(("Copyright Policy")));        
    $this->selenium->isElementPresent("xpath=//p[@class='licenseText']/a");
    $this->selenium->click("xpath=//p[@class='licenseText']/a[2]");
    $this->selenium->selectWindow("_blank");    
    $this->selenium->waitForPageToLoad(10000);    
    $this->assertTrue($this->selenium->isTextPresent(("Directive 2001/29/EC of the European Parliament and of the Council")));
    $this->assertTrue($this->selenium->isTextPresent(("32001L0029")));
    $this->selenium->close();
    $this->selenium->selectWindow(null);
    
    $this->selenium->click("xpath=//p[@class='licenseText']/a[3]");
    $this->selenium->selectWindow("_blank");    
    $this->selenium->waitForPageToLoad(10000);    
    $this->assertTrue($this->selenium->isTextPresent(("Chilling Effects")));
    $this->assertTrue($this->selenium->isTextPresent(("Chilling Effects Clearinghouse - www.chillingeffects.org")));
    $this->selenium->close();
    $this->selenium->selectWindow(null);
    
    $this->selenium->goBack();
    $this->selenium->waitForPageToLoad(10000);
    // imprint
    $this->selenium->click("xpath=//a[@class='license'][4]");    
    $this->selenium->waitForPageToLoad(10000);
    $this->assertTrue($this->selenium->isTextPresent(("Address")));
    $this->assertTrue($this->selenium->isTextPresent(("Institut für Softwaretechnologie")));
    $this->assertTrue($this->selenium->isTextPresent(("Technische Universität Graz")));
    $this->assertTrue($this->selenium->isTextPresent(("Inffeldgasse 16B/II")));
    $this->assertTrue($this->selenium->isTextPresent(("8010 Graz")));
    $this->assertTrue($this->selenium->isTextPresent(("Austria")));
    $this->selenium->click("xpath=//p[@class='licenseText']/a");    
    $this->selenium->selectWindow("_blank");    
    $this->selenium->waitForPageToLoad(10000);
    $this->assertRegExp("/IST web - Index/", $this->selenium->getTitle());  
    $this->selenium->close();
    $this->selenium->selectWindow(null);   
    $this->selenium->goBack();    
    $this->selenium->waitForPageToLoad(10000);
  
     // privacy policy
    $this->selenium->click("xpath=//a[@class='license']");    
    $this->selenium->waitForPageToLoad(10000);
    $this->assertTrue($this->selenium->isTextPresent(("Privacy Policy")));        
    $this->selenium->isElementPresent("xpath=//p[@class='licenseText']/a");
    $this->selenium->goBack();    
    $this->selenium->waitForPageToLoad(10000);

    
    // test terms of use
    $this->selenium->click("xpath=//a[@class='license'][2]");    
    $this->selenium->waitForPageToLoad(10000);
    $this->assertTrue($this->selenium->isTextPresent(("Welcome to the Catroid community! As part of the Catroid community, you are sharing projects and ideas with people: ")));    
    $this->selenium->click("xpath=//p[@class='licenseText'][3]/a");    
    $this->selenium->selectWindow("_blank");    
    $this->selenium->waitForPageToLoad(10000);
    $this->assertRegExp("/Creative Commons — Attribution-ShareAlike 2.0 Generic — CC BY-SA 2.0/", $this->selenium->getTitle());  
    $this->selenium->close();
    $this->selenium->selectWindow(null);
    
    $this->selenium->click("xpath=//p[@class='licenseText']/a[2]");    
    $this->selenium->selectWindow("_blank");    
    $this->selenium->waitForPageToLoad(10000);    
    $this->assertTrue($this->selenium->isTextPresent("GNU GENERAL PUBLIC LICENSE"));
    $this->assertTrue($this->selenium->isTextPresent("Version 3, 29 June 2007"));
    $this->selenium->close();
    $this->selenium->selectWindow(null);

    $this->selenium->click("xpath=//p[@class='licenseText']/a[3]");    
    $this->selenium->selectWindow("_blank");    
    $this->selenium->waitForPageToLoad(10000);    
    $this->assertTrue($this->selenium->isTextPresent("GNU AFFERO GENERAL PUBLIC LICENSE"));
    $this->assertTrue($this->selenium->isTextPresent("Version 3, 19 November 2007"));
    $this->selenium->close();
    $this->selenium->selectWindow(null);
    
    $this->selenium->click("xpath=//p[@class='licenseText']/a[4]");
    $this->selenium->selectWindow("_blank");
    $this->selenium->waitForPageToLoad(10000);
    $this->assertRegExp("/catroid \-/", $this->selenium->getTitle());  
    $this->assertRegExp("/An on-device graphical programming language for Android inspired by Scratch/", $this->selenium->getTitle());
    $this->selenium->close();
    $this->selenium->selectWindow(null);
    $this->selenium->goBack();    
            
    
  }
}
?>

File: tests/selenium-grid/src/catroid/LicenseTests.java
/*    Catroid: An on-device graphical programming language for Android devices
 *    Copyright (C) 2010-2011 The Catroid Team
 *    (<http://code.google.com/p/catroid/wiki/Credits>)
 *
 *    This program is free software: you can redistribute it and/or modify
 *    it under the terms of the GNU Affero General Public License as
 *    published by the Free Software Foundation, either version 3 of the
 *    License, or (at your option) any later version.
 *
 *    This program is distributed in the hope that it will be useful,
 *    but WITHOUT ANY WARRANTY; without even the implied warranty of
 *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *    GNU Affero General Public License for more details.
 *
 *    You should have received a copy of the GNU Affero General Public License
 *    along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */

package at.tugraz.ist.catroweb.catroid.license;

import static com.thoughtworks.selenium.grid.tools.ThreadSafeSeleniumSessionStorage.closeSeleniumSession;
import static com.thoughtworks.selenium.grid.tools.ThreadSafeSeleniumSessionStorage.session;
import static com.thoughtworks.selenium.grid.tools.ThreadSafeSeleniumSessionStorage.startSeleniumSession;
import com.thoughtworks.selenium.Selenium;
import static org.testng.AssertJUnit.assertTrue;
import org.testng.annotations.AfterMethod;
import org.testng.annotations.BeforeMethod;
import org.testng.annotations.Parameters;
import org.testng.annotations.Test;

import at.tugraz.ist.catroweb.common.*;

public class LicenseTests {
  protected ExternalLinkTester externalLinkTester;
  
  @BeforeMethod(groups = {"default", "license"}, alwaysRun = true)
  @Parameters({"seleniumHost", "seleniumPort", "browser", "webSite"})  
  protected void startSession(String seleniumHost, int seleniumPort, String browser, String webSite) {
    startSeleniumSession(seleniumHost, seleniumPort, browser, webSite);
    session().setSpeed(CommonFunctions.setSpeed());
    session().setTimeout(CommonConfig.TIMEOUT);
    
    externalLinkTester = new ExternalLinkTester(seleniumHost, seleniumPort, browser);
  }

  @AfterMethod(groups = {"default", "license"}, alwaysRun = true)
  protected void closeSession() {
    externalLinkTester.cleanup();
    closeSeleniumSession();
  }

  @Test(groups = {"license", "firefox", "default"}, description = "check privacy policy link/page")
  public void privacyPolicy() throws Throwable {
    session().open(CommonConfig.TESTS_BASE_PATH);
    session().waitForPageToLoad(CommonConfig.TIMEOUT);
    session().click("xpath=//a[@class='license']");
    session().waitForPageToLoad(CommonConfig.TIMEOUT);

    assertTrue(session().isTextPresent("Privacy Policy"));
    session().isElementPresent("xpath=//p[@class='licenseText']/a");
  }

  @Test(groups = {"license", "firefox", "default"}, description = "check terms of use link/page")
  public void termsOfUse() throws Throwable {
	session().open(CommonConfig.TESTS_BASE_PATH);
    session().waitForPageToLoad(CommonConfig.TIMEOUT);
    session().click("xpath=//a[@class='license'][2]");
    session().waitForPageToLoad(CommonConfig.TIMEOUT);

    assertTrue(session().isTextPresent("Welcome to the Catroid community!"));
    assertTrue(session().isTextPresent("As part of the Catroid community, you are sharing projects and ideas with people:"));
    Selenium popupSession = externalLinkTester.getSession(session(), "xpath=//p[@class='licenseText'][3]/a");
    CommonAssertions.assertRegExp(".*Creative Commons — Attribution-ShareAlike 2.0 Generic — CC BY-SA 2.0.*", popupSession.getTitle());
    externalLinkTester.stopSession(popupSession);

    popupSession = externalLinkTester.getSession(session(), "xpath=//p[@class='licenseText']/a[2]");
    assertTrue(popupSession.isTextPresent("GNU GENERAL PUBLIC LICENSE"));
    assertTrue(popupSession.isTextPresent("Version 3, 29 June 2007"));
    externalLinkTester.stopSession(popupSession);

    popupSession = externalLinkTester.getSession(session(), "xpath=//p[@class='licenseText']/a[3]");
    assertTrue(popupSession.isTextPresent("GNU AFFERO GENERAL PUBLIC LICENSE"));
    assertTrue(popupSession.isTextPresent("Version 3, 19 November 2007"));
    externalLinkTester.stopSession(popupSession);

    popupSession = externalLinkTester.getSession(session(), "xpath=//p[@class='licenseText']/a[4]");
    CommonAssertions.assertRegExp(".*catroid -.*", popupSession.getTitle());
    CommonAssertions.assertRegExp(".*An on-device graphical programming language for Android inspired by Scratch.*", popupSession.getTitle());
    externalLinkTester.stopSession(popupSession);
  }

  @Test(groups = {"license", "firefox", "default"}, description = "check copyright policy link/page")
  public void copyrightPolicy() throws Throwable {
    session().open(CommonConfig.TESTS_BASE_PATH);
    session().waitForPageToLoad(CommonConfig.TIMEOUT);
    session().click("xpath=//a[@class='license'][3]");
    session().waitForPageToLoad(CommonConfig.TIMEOUT);

    assertTrue(session().isTextPresent("Copyright Policy"));
    session().isElementPresent("xpath=//p[@class='licenseText']/a");
    Selenium popupSession = externalLinkTester.getSession(session(), "xpath=//p[@class='licenseText']/a[2]");
    assertTrue(popupSession.isTextPresent("Directive 2001/29/EC of the European Parliament and of the Council"));
    assertTrue(popupSession.isTextPresent("32001L0029"));
    externalLinkTester.stopSession(popupSession);

    popupSession = externalLinkTester.getSession(session(), "xpath=//p[@class='licenseText']/a[3]");
    assertTrue(popupSession.isTextPresent("Chilling Effects"));
    assertTrue(popupSession.isTextPresent("Chilling Effects Clearinghouse - www.chillingeffects.org"));
    externalLinkTester.stopSession(popupSession);
  }

  @Test(groups = {"license", "firefox", "default"}, description = "check imprint link/page")
  public void imprint() throws Throwable {
    session().open(CommonConfig.TESTS_BASE_PATH);
    session().waitForPageToLoad(CommonConfig.TIMEOUT);
    session().click("xpath=//a[@class='license'][4]");
    session().waitForPageToLoad(CommonConfig.TIMEOUT);

    assertTrue(session().isTextPresent("Address"));
    assertTrue(session().isTextPresent("Institut für Softwaretechnologie"));
    assertTrue(session().isTextPresent("Technische Universität Graz"));
    assertTrue(session().isTextPresent("Inffeldgasse 16B/II"));
    assertTrue(session().isTextPresent("8010 Graz"));
    assertTrue(session().isTextPresent("Austria"));
    Selenium popupSession = externalLinkTester.getSession(session(), "xpath=//p[@class='licenseText']/a");
    CommonAssertions.assertRegExp(".*IST web - Index.*", popupSession.getTitle());
    externalLinkTester.stopSession(popupSession);
  }

  @Test(groups = {"license", "firefox", "default"}, description = "check contact us link/page")
  public void contactUs() throws Throwable {
    session().open(CommonConfig.TESTS_BASE_PATH);
    session().waitForPageToLoad(CommonConfig.TIMEOUT);
    session().click("xpath=//a[@class='license'][5]");
    session().waitForPageToLoad(CommonConfig.TIMEOUT);
    assertTrue(session().isTextPresent(("Contact us")));
    session().isElementPresent("xpath=//p[@class='licenseText']/a");
  }
}
